### ISSUE TITLE
Code Fix for NUnit2046 Reported For Span Results In Error

### ISSUE THREAD

**Issue by JasonBock** [nunit/nunit.analyzers#474]
To reproduce this, create a class library for .NET 6 with the following NUnit references:
```
<PackageReference Include="NUnit" Version="3.13.3" />
<PackageReference Include="NUnit.Analyzers" Version="3.3.0" />
```
Create the following test:
```csharp
[Test]
public static void TestSpan()
{
  var span = new byte[8].AsSpan();
  Assert.That(span.Length, Is.EqualTo(8));
}
```
NUnit2046 will light up for the `Assert`. However, if I use the "Use Has.Length" code fix:

![image](https://user-images.githubusercontent.com/904213/189156403-3ed5e1f7-506e-4e6d-8819-fcc074c3ec5e.png)

This creates the following compiler error:
```
CS0306 The type 'Span<byte>' may not be used as a type argument
```

**Comment by mikkelbu** [comment]
Thanks for the report @JasonBock. I guess we should not report and supply a code fix for types that are ref structs.

**Comment by manfred-brands** [comment]
I will fix it.

**Comment by manfred-brands** [comment]
Actually the latest source no longer triggers the suggestion for Span, because of changes done for #459 (PR #460).
This stops the suggestion on types that do not implement `IEnumerable`, which includes Span.
When trying to create a `ref struct` that implements `IEnumerable`, I get the warning:
```
error CS8343: 'EnumerableRef': ref structs cannot implement interfaces
```

So I think all we need is a new release?

**Comment by mikkelbu** [comment]
This was closed in release 3.5
